File: src/services/booking.service.js
import api from './api'

const bookingService = {
  // Public/Customer - Create a new booking
  createBooking: async (bookingData) => {
    const response = await api.post('/bookings', bookingData)
    return response.data.data
  },

  // Customer/Admin - Get all bookings (Backend checks role via token)
  getAllBookings: async () => {
    const response = await api.get('/bookings')
    return response.data.data
  },

  // Customer/Admin - Get single booking details
  getBookingById: async (id) => {
    const response = await api.get(`/bookings/${id}`)
    return response.data.data
  },

  // Admin - Update booking (e.g., Status to CONFIRMED or COMPLETED)
  updateBooking: async (id, bookingData) => {
    const response = await api.put(`/bookings/${id}`, bookingData)
    return response.data.data
  },

  // Admin - Delete booking
  deleteBooking: async (id) => {
    const response = await api.delete(`/bookings/${id}`)
    return response.data.data || response.data
  },

  // Customer/Admin - Cancel booking request
  cancelBooking: async (id) => {
    const response = await api.put(`/bookings/${id}/cancel`)
    return response.data.data || response.data
  }
}

export default bookingService

File: OneDrive/Desktop/EventBookingApp/src/main/java/com/ranadj/service/BookingService.java
package com.ranadj.service;

import com.ranadj.dto.BookingRequest;
import com.ranadj.dto.BookingResponse;
import com.ranadj.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/**
 * Service interface specifying Booking operations.
 */
public interface BookingService {

    /**
     * Creates a new booking.
     */
    BookingResponse createBooking(BookingRequest request, User currentUser);

    /**
     * Updates an existing booking.
     */
    BookingResponse updateBooking(Long id, BookingRequest request, User currentUser);

    /**
     * Cancels a booking (marks status as CANCELLED).
     */
    void cancelBooking(Long id, User currentUser);

    /**
     * Retrieves a booking by its ID, with owner checks.
     */
    BookingResponse getBookingById(Long id, User currentUser);

    /**
     * Retrieves paginated bookings.
     * Customers will only see their own bookings; Admins will see all.
     */
    Page<BookingResponse> getAllBookings(Pageable pageable, User currentUser);

    /**
     * Deletes a booking from the database. Admin only.
     */
    void deleteBooking(Long id);
}
